src/include/sof/drivers/sai.h

```diff
@@ -0,0 +1,243 @@
+#define REG_SAI_CR3_TRCE_MASK	MASK(23, 16)
+#define REG_SAI_CR3_TRCE(x)	(x << 16)
+#define REG_SAI_CR3_WDFL(x)	(x)
+#define REG_SAI_CR3_WDFL_MASK	0x1f
+
+/* SAI Transmit and Receive Configuration 4 Register */
+
+#define REG_SAI_CR4_FCONT	BIT(28)
+#define REG_SAI_CR4_FCOMB_SHIFT BIT(26)
+#define REG_SAI_CR4_FCOMB_SOFT  BIT(27)
+#define REG_SAI_CR4_FCOMB_MASK	MASK(27, 26)
+#define REG_SAI_CR4_FPACK_8     (0x2 << 24)
+#define REG_SAI_CR4_FPACK_16    (0x3 << 24)
+#define REG_SAI_CR4_FRSZ(x)	(((x) - 1) << 16)
+#define REG_SAI_CR4_FRSZ_MASK	MASK(20, 16)
+#define REG_SAI_CR4_SYWD(x)	(((x) - 1) << 8)
+#define REG_SAI_CR4_SYWD_MASK	MASK(12, 8)
+#define REG_SAI_CR4_CHMOD	BIT(5)
+#define REG_SAI_CR4_CHMOD_MASK	BIT(5)
+#define REG_SAI_CR4_MF		BIT(4)
+#define REG_SAI_CR4_FSE		BIT(3)
+#define REG_SAI_CR4_FSP		BIT(1)
+#define REG_SAI_CR4_FSD_MSTR	BIT(0)
+
+/* SAI Transmit and Receive Configuration 5 Register */
+#define REG_SAI_CR5_WNW(x)	(((x) - 1) << 24)
+#define REG_SAI_CR5_WNW_MASK	MASK(28, 24)
+#define REG_SAI_CR5_W0W(x)	(((x) - 1) << 16)
+#define REG_SAI_CR5_W0W_MASK	MASK(20, 16)
+#define REG_SAI_CR5_FBT(x)	(((x) - 1) << 8)
+#define REG_SAI_CR5_FBT_MASK	MASK(12, 8)
+
+/* SAI MCLK Control Register */
+#define REG_SAI_MCTL_MCLK_EN	BIT(30)	/* MCLK Enable */
+#define REG_SAI_MCTL_MSEL_MASK	MASK(25, 24)
+#define REG_SAI_MCTL_MSEL(id)   ((id) << 24)
+#define REG_SAI_MCTL_MSEL_BUS	SET_BITS(25, 24, 0)
+#define REG_SAI_MCTL_MSEL_MCLK1	SET_BITS(25, 24, 1)
+#define REG_SAI_MCTL_MSEL_MCLK2	SET_BITS(25, 24, 2)
+#define REG_SAI_MCTL_MSEL_MCLK3	SET_BITS(25, 24, 3)
+#define REG_SAI_MCTL_DIV_EN	BIT(23)
+#define REG_SAI_MCTL_DIV_MASK	0xFF
+
+/* SAI VERID Register */
+#define REG_SAI_VER_ID_SHIFT	16
+#define REG_SAI_VER_ID_MASK	MASK(REG_SAI_VER_ID_SHIFT + 15,\
+		REG_SAI_VER_ID_SHIFT)
+#define REG_SAI_VER_EFIFO_EN	BIT(0)
+#define REG_SAI_VER_TSTMP_EN	BIT(1)
+
+/* SAI PARAM Register */
+#define REG_SAI_PAR_SPF_SHIFT	16
+#define REG_SAI_PAR_SPF_MASK	MASK(REG_SAI_PAR_SPF_SHIFT + 3,\
+		REG_SAI_PAR_SPF_SHIFT)
+#define REG_SAI_PAR_WPF_SHIFT	8
+#define REG_SAI_PAR_WPF_MASK	MASK(REG_SAI_PAR_WPF_SHIFT + 3,\
+		REG_SAI_PAR_WPF_SHIFT)
+#define REG_SAI_PAR_DLN_MASK	(0x0F)
+
+/* SAI MCLK Divide Register */
+#define REG_SAI_MDIV_MASK	0xFFFFF
+
+/* SAI type */
+#define REG_SAI_DMA		BIT(0)
+#define REG_SAI_USE_AC97	BIT(1)
+#define REG_SAI_NET		BIT(2)
+#define REG_SAI_TRA_SYN		BIT(3)
+#define REG_SAI_REC_SYN		BIT(4)
+#define REG_SAI_USE_I2S_SLAVE	BIT(5)
+
+#define REG_FMT_TRANSMITTER	0
+#define REG_FMT_RECEIVER	1
+
+/* SAI clock sources */
+#define REG_SAI_CLK_BUS		0
+#define REG_SAI_CLK_MAST1	1
+#define REG_SAI_CLK_MAST2	2
+#define REG_SAI_CLK_MAST3	3
+
+#define REG_SAI_MCLK_MAX	4
+
+/* SAI data transfer numbers per DMA request */
+#define REG_SAI_MAXBURST_TX 6
+#define REG_SAI_MAXBURST_RX 6
+
+#define SAI_FLAG_PMQOS   BIT(0)
+
+#define SAI_IRQ			348
+#define EDMA_SAI_IRQ		349
+
+#define EDMA_SAI_RX_CHAN	14
+#define EDMA_SAI_TX_CHAN	15
+
```

Comment:
Please remove this and use the ones defined in edma.h

Reply:
To get rid of #define EDMA_SAI_RX_CHAN	14 I will need to rebase to master my branch